bench-run/src/main/scala/dotty/tools/benchmarks/specialization/Functions.scala

```diff
@@ -0,0 +1,71 @@
+package dotty.tools.benchmarks.specialization
+
+import org.openjdk.jmh.annotations._
+import scala.util.Random
+
+
+@State(Scope.Benchmark)
+class Functions {
+  extension (x: Int)
+    inline def times(inline work: Int): Int = {
+      var res = 0
+      var count = 0
+      while count < x do
```

Comment:
I haven't checked what is being done here exactly but note that it's not recommended to use loops when benchmarking on the JVM: http://tutorials.jenkov.com/java-performance/jmh.html#writing-good-benchmarks, instead let JMH run things as many times as needed to get good statistics

Reply:
Thanks for the reminder, @smarter . It's really subtle to defeat optimization in microbenchmarks.